File: app/src/main/java/com/fastaccess/data/service/ImgurService.java
package com.fastaccess.data.service;

import android.support.annotation.Nullable;

import com.fastaccess.data.dao.ImgurReponseModel;

import okhttp3.RequestBody;
import retrofit2.http.Body;
import retrofit2.http.POST;
import retrofit2.http.Query;
import io.reactivex.Observable;

/**
 * Created by Kosh on 15 Apr 2017, 8:06 PM
 */

public interface ImgurService {
    @POST("image")
    Observable<ImgurReponseModel> postImage(@Nullable @Query("title") String title, @Body RequestBody body);
}

File: domain/src/main/java/com/fastaccess/domain/repository/services/ImgurService.kt
package com.fastaccess.domain.repository.services

import com.fastaccess.domain.response.ImgureResponseModel
import io.reactivex.Observable
import okhttp3.RequestBody
import retrofit2.http.Body
import retrofit2.http.POST
import retrofit2.http.Query


/**
 * Created by Kosh on 2019-07-29.
 */
interface ImgurService {
    @POST("image")
    fun postImage(@Query("title") title: String, @Body body: RequestBody): Observable<ImgureResponseModel>

}